### PR TITLE
feat: Upgrade Flutter version

### PR DESCRIPTION
**Pull request by internetova**
## Changes
* Updates minimum Flutter version to 3.0.0
* Updates minimum surf_lint_rules version to 3.0.0
*  Fixed dart analysis

## Checklist for self-check
- [x] Commits and PRs have been filed according to [the rules on the project](https://github.com/surfstudio/surf-flutter-app-template#workflow-in-a-repository).
- [x] The author is marked as an assigne and assigned mandatory reviewers.
- [ ] Required labels marked
- [ ] Specified related tasks and/or related PRs.
- [ ] Specified Changes.
- [ ] Attached videos/screenshots demonstrating the fix/feature.
- [ ] All unspecified fields in the PR description deleted.
- [ ] New code covered by tests.

## Checklist for reviewers
- [ ] CI passed successfully _(with a green check mark)_.
- [ ] PR is atomic, by volume no more than 400 (+-) corrected lines (not including codogen).

Design:
- [ ] System design corresponds to the agreements on structure and architecture on the project.
- [ ] The code is decomposed into necessary and sufficient components.

Functionality:
- [ ] The code solves the problem.
- [ ] Any changes to the user interface are reasonable and look good.

Complexity:
- [ ] The code is clear, easy to read, functions are small, no more than 50 lines.
- [ ] The logic is not overcomplicated, there is no overengineering (no code sections that may be needed in the future, but no one knows about it).

Tests:
- [ ] Updated or added tests for mandatory components.
- [ ] The tests are correct, helpful, and well designed/developed.

Naming:
- [ ] The naming of variables, methods, classes and other components is understandable.

Comments:
- [ ] The comments are understandable and helpful.

Documentation:
- [ ] All labels are correct
- [ ] Technical documentation updated (after approval, updates last reviewer).

### CODE REVIEW

File: example/lib/bottom_sheet_based_on_content_height.dart [hunk 1]
```diff
@@ -43,18 +42,18 @@ class _AnimalCard extends StatelessWidget {
   Widget build(BuildContext context) {
```
Comment (ViktorJasch) [inline example/lib/bottom_sheet_based_on_content_height.dart:42]: К проекту подключен member-ordering линтер, который ругается на этот метод. Вижу что изменения были сделаны не в рамках этой задачи, но возможно, пока изменений немного, стоит пофиксить в этом месте замечания линтера

**Review by ViktorJasch — APPROVED** [review]

File: test/sticky_bottom_sheet_test.dart [hunk 2]
```diff
@@ -1,10 +1,10 @@
 // Copyright (c) 2019-present,  SurfStudio LLC
 //
-// Licensed under the Apache License, Version 2.0 (the "License");
+// Licensed under the Apache License, Version 2 (the "License");
 // you may not use this file except in compliance with the License.
 // You may obtain a copy of the License at
 //
-//     http://www.apache.org/licenses/LICENSE-2.0
```
Comment (alexey-fetisov) [inline test/sticky_bottom_sheet_test.dart:7]: В лицензии апача в оригинале  2.0, а не 2. И в ссылках и остальных местах, где прописана версия.

File: test/bottom_sheet_test.dart [hunk 3]
```diff
@@ -1,19 +1,17 @@
 // Copyright (c) 2019-present,  SurfStudio LLC
 //
-// Licensed under the Apache License, Version 2.0 (the "License");
+// Licensed under the Apache License, Version 2 (the "License");
 // you may not use this file except in compliance with the License.
 // You may obtain a copy of the License at
 //
-//     http://www.apache.org/licenses/LICENSE-2.0
+//     http://www.apache.org/licenses/LICENSE-2
```
Comment (alexey-fetisov) [inline test/bottom_sheet_test.dart:7]: В лицензии апача в оригинале 2.0, а не 2. И в ссылках и остальных местах, где прописана версия.

**Review by mpkander — APPROVED** [review]

File: test/bottom_sheet_test.dart (hunk 3 above)
Comment (mpkander) [reply test/bottom_sheet_test.dart:7]: @internetova Need to fix

**Review by mpkander — CHANGES_REQUESTED** [review]
Fix the license issue

**Pull request merged**